src/App.js

```diff
@@ -10,25 +10,30 @@ class App extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      authorization: {}
+      authenticationValues: {}
     }
   }
 
   render() {
     return (
       <MuiThemeProvider>
         <div className="App">
-          <LoginWithAmazon setAuthorization={(authResponse) =>
-            this.setState({authorization: {access_token: authResponse.access_token, expires_in: authResponse.expires_in}})} />
+          <LoginWithAmazon setAuthorization={(authResponse) => this.setAuthenticationValues(authResponse)} />
 
           {/* This will not be rendered until access_token and expires_in are available in the state.
            TODO: Replace this with routing logic i.e., route to ChatWindow on successful login */}
-          { this.state && this.state.authorization.access_token && this.state.authorization.expires_in &&
-            <ChatWindow authorization={this.state.authorization}/> }
+          { this.state && this.state.authenticationValues.access_token && this.state.authenticationValues.expires_in &&
+            <ChatWindow authenticationValues={this.state.authenticationValues}/> }
         </div>
       </MuiThemeProvider>
     );
   }
+
+  setAuthenticationValues(authResponse) {
+    if(authResponse !== undefined) {
+      this.setState({authenticationValues: {access_token: authResponse.access_token, expires_in: authResponse.expires_in}})
```

Comment:
Don't you want to validate the authResponse that it contains all mandatory fields and populate the state object only if everything is as expected?
Otherwise, every component that needs to use the authInfo in the state needs to do its own validation.

Reply:
authResponse is a response object provided by LoginWithAmazon. Their contract is to provide access_token and expires_in on success and this function is to handle event success.
Reference: https://developer.amazon.com/docs/login-with-amazon/implicit-grant.html#authorization-response